traits/ctraits.pyx

```diff
@@ -1370,15 +1370,8 @@ cdef object getattr_delegate(cTrait trait, CHasTraits obj, object name):
         delegate_attr_name = trait.delegate_attr_name(trait, obj, name)
         tp = type(delegate)
 
-        try:
-            result = getattr(delegate, delegate_attr_name)
-            return result
-        except NotImplementedError as exc:
-            raise exc
-        except Exception as exc:
-            raise DelegationError("The '%.50s' object has no attribute "
```

Comment:
@jvkersch The change in this PR removes the usage of the `DelegationError` for a specific case. I'm not sure if the change in behaviour is desired for all the other cases. I think you'll need to handle and filter `AttributeError`s raised for delegated events to solve the problem, no?